Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -2443,6 +2443,30 @@ theorem sup_mul_nat (o : Ordinal) : (sup fun n : ℕ => o * n) = o * ω := by
     exact sup_eq_zero_iff.2 fun n => zero_mul (n : Ordinal)
   · exact (mul_isNormal ho).apply_omega0
 
+/-- The order isomorphism between ℕ and the first ω ordinals. -/
+def relIso_nat_omega : ℕ ≃o Iio ω where
+  toFun n := ⟨n, nat_lt_omega0 n⟩
+  invFun n := Classical.choose (lt_omega0.1 n.2)
+  left_inv n := by
+    have h : ∃ m : ℕ, n = (m : Ordinal) := ⟨n, rfl⟩
+    exact (Nat.cast_inj.1 (Classical.choose_spec h)).symm
+  right_inv n := Subtype.eq (Classical.choose_spec (lt_omega0.1 n.2)).symm
+  map_rel_iff' := fun {n m} ↦ by simp only [coe_fn_mk, Subtype.mk_le_mk, Nat.cast_le]
+
+theorem relIso_nat_omega.symm_eq {o : Ordinal} (h : o < ω) :
+    relIso_nat_omega.symm ⟨o, h⟩ = o := by
+  rcases lt_omega0.mp h with ⟨n, rfl⟩
+  exact congrArg Nat.cast <| relIso_nat_omega.symm_apply_apply n
+
+theorem strictMono_of_succ_lt_omega {o : Ordinal} (f : Iio ω → Iio o)
+    (hf : ∀ i, f i < f ⟨succ i, isLimit_omega0.succ_lt i.2⟩) {i j} (h : i < j) : f i < f j := by
+  have mono := strictMono_nat_of_lt_succ fun n ↦ hf ⟨n, nat_lt_omega0 n⟩
+  have := mono <| (OrderIso.lt_iff_lt relIso_nat_omega.symm).mpr h
+  change f ⟨relIso_nat_omega.symm ⟨i.1, i.2⟩, _⟩ <
+    f ⟨relIso_nat_omega.symm ⟨j.1, j.2⟩, _⟩ at this
+  simp_rw [relIso_nat_omega.symm_eq] at this
+  exact this
```

Comment:
```suggestion
theorem strictMono_of_succ_lt_omega {α : Type*} [Preorder α] (f : Iio ω → α)
    (hf : ∀ i, f i < f ⟨succ i, isLimit_omega0.succ_lt i.2⟩) : StrictMono f := by
  have mono := strictMono_nat_of_lt_succ fun n ↦ hf ⟨n, nat_lt_omega0 n⟩
  convert mono.comp relIso_nat_omega.symm.strictMono
  ext
  simp
```
You could also add a version proving `Monotone f`.

Reply:
That's just `↑.monotone` though.

Reply:
The assumption would be `f x ≤ f (succ x)` instead, so not quite.

Reply:
```lean4
theorem monotone_of_succ_lt_omega0 {α : Type*} [Preorder α] (f : Iio ω → α)
    (hf : ∀ i, f i ≤ f ⟨succ i, isLimit_omega0.succ_lt i.2⟩) : Monotone f := sorry
```

Reply:
Oh I misunderstood.